### PR TITLE
Fix leaks after calling wasm drop

### LINKED ISSUE #6505: Using wasm `drop` method results in warnings flood
This is an issue to fix leftovers after implementing `wasm.drop` method. Original problem was rised during QAing https://github.com/enso-org/enso/pull/6365#issuecomment-1521870772

Steps to reproduce:
1. Open Enso with `-debug.dev-tools` flag
2. Navigate to console pane
3. Type `ensoglApp.wasm.drop()`
4. Warnings will flood both console and terminal

https://drive.google.com/file/d/1Oy3-D5JGYy4PbJhYAXYXB3Ouuks_zL45/view?usp=share_link

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

1. Fixed leaks of Scene. We were making several cycles of Rc references.
2. Fix removing event listeners: the remove listener option should take same options as those passed in addEventListener.

This should finally fix #6505 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- ~~[ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: lib/rust/ensogl/core/src/control/io/mouse.rs [hunk 1]
```diff
@@ -111,15 +111,13 @@ macro_rules! define_bindings {
...
+    // 1. We listen for events in the bubbling phase. If we ever would like to listen in the capture
     // phase, it would need to be set to "bubbling" for the "mouseleave" and "mouseenter" events,
     // as they provide incorrect events for the "capture" phase.
-    options.capture(false);
-    // We want to prevent default action on wheel events, thus listener cannot be passive.
-    options.passive(false);
-    options
+    //
+    // 2. We want to prevent default action on wheel events, thus listener cannot be passive.
+    web::EventListenerHandleOptions::new().passive()
```
Comment (kazcw) [inline lib/rust/ensogl/core/src/control/io/mouse.rs]: Should this be `not_passive()`?

File: lib/rust/frp/src/io/js.rs [hunk 2]
```diff
@@ -69,9 +69,7 @@ impl Listener {
 
 /// Retrun options for addEventListener function. See also
 /// https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
-fn event_listener_options() -> enso_web::AddEventListenerOptions {
-    let mut options = enso_web::AddEventListenerOptions::new();
+fn event_listener_options() -> enso_web::EventListenerHandleOptions {
     // We want to prevent default action on wheel events, thus listener cannot be passive.
-    options.passive(false);
-    options
+    enso_web::EventListenerHandleOptions::new().passive()
```
Comment (kazcw) [inline lib/rust/frp/src/io/js.rs]: `not_passive`?

**Review by kazcw — APPROVED** [review]

File: lib/rust/web/src/lib.rs [hunk 3]
```diff
@@ -721,6 +721,83 @@ pub fn ignore_context_menu(target: &EventTarget) -> EventListenerHandle {
...
+    pub fn passive(mut self) -> Self {
+        self.passive = Some(true);
+        self
+    }
+
+    pub fn not_passive(mut self) -> Self {
+        self.passive = Some(false);
+        self
+    }
+
```
Comment (wdanilo) [inline lib/rust/web/src/lib.rs]: Please add here docs. It's not obvious what these do after reading the name if someone (like me) do not remember what the JS side does.

File: lib/rust/web/src/lib.rs [hunk 4]
```diff
@@ -759,46 +837,64 @@ struct EventListenerHandleData {
...
+            .remove_event_listener_with_callback_and_event_listener_options(
+                &self.name,
+                function,
+                &self.options.into(),
+            )
+            .ok();
+    }
+}
+
+
```
Comment (wdanilo) [inline lib/rust/web/src/lib.rs]: too big spacing

**Review by wdanilo — APPROVED** [review]

**Comment by kazcw** [comment]
# QA: :green_circle:

**Pull request merged**